### ISSUE TITLE
Add command to output cargo config values

### ISSUE THREAD

**Issue by DCjanus** [rust-lang/cargo#7075]
**Describe the problem you are trying to solve**
I'm trying to add feature for [cargo-edit](https://github.com/killercup/cargo-edit), a cargo plugin.

I want to query crate versions from local registry index rather than HTTP API, but I don't want to depend on cargo as a lib (because it's huge). So I have to parse cargo config and get registry index path.

Here is the problem: directory name of registry index looks like `{hostname}-{hash}`, and part of the hash implement is [here](https://github.com/rust-lang/cargo/blob/83d086da9467edb969d25a1223a902815388eb8c/src/cargo/core/source/source_id.rs#L496-L504), which means I have to introduce the type in cargo, otherwise I can't get the hash.

**Describe the solution you'd like**
That's would be great if there is a subcommand to output some cargo config values, just like `cargo-metadata`, but for cargo itself.

With this, I could just read the JSON of the command output, that's would be very helpful.

Image in my mind:

```bash
$ cargo config --list --format-version=1
{
    "sources":[
        { "name": "crates-io", "registry": "https://github.com/rust-lang/crates.io-index", "index_path": "/home/dcjanus/.cargo/registry/index/github.com-1ecc6299db9ec823", "replace-with": "rustcc" },
        { "name": "rustcc", "registry": "https://code.aliyun.com/rustcc/crates.io-index.git", "index_path": "/home/dcjanus/.cargo/registry/index/code.aliyun.com-738b7dba08a2a41e" }
    ]
    ...
}
```

**Comment by ehuss** [comment]
I don't think we ever want to expose the `.cargo/registry` paths, as these are internal to cargo and not something we want to support for external tools.

**Comment by ehuss** [comment]
`cargo config` has been implemented on nightly and is tracked in #9301.
Providing access to the index is tracked in #7824 (though as noted there, there are issues with assuming cargo will have access to the whole index).
I'm going to close in favor of those issues.
